src/modules/extrakeys/main_impl.h

```diff
@@ -186,6 +186,16 @@ int secp256k1_keypair_create(const secp256k1_context* ctx, secp256k1_keypair *ke
     return ret;
 }
 
+int secp256k1_keypair_seckey(const secp256k1_context* ctx, unsigned char *seckey32, const secp256k1_keypair *keypair) {
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(seckey32 != NULL);
+    memset(seckey32, 0, 32);
+    ARG_CHECK(keypair != NULL);
+
+    memcpy(seckey32, &keypair->data[0], 32);
```

Comment:
You may want to use` secp256k1_keypair_seckey_load` instead of memcpy because it correctly handles zeroed keypairs.

Reply:
I change it, I've used `memcpy` because just underneath the function that extracts a pubkey just does a mempy:
https://github.com/bitcoin-core/secp256k1/blob/e89278f211a526062745c391d48a7baf782b4b2b/src/modules/extrakeys/main_impl.h#L195

Reply:
And I'll add more tests

Reply:
I see. Yeah, it should be consistent with the pubkey getter. Fine for now imo, since this outputs an invalid seckey, we don't mention a guarantee that a function fails if called with an invalid keypair and the current version is simpler than the alternative.